src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsEntityTypeDefinition.cs

```diff
@@ -112,7 +112,39 @@ private IEnumerable<IEdmStructuralProperty> ComputeDeclaredKey()
                 List<IEdmStructuralProperty> key = new List<IEdmStructuralProperty>();
                 foreach (CsdlPropertyReference keyProperty in this.entity.Key.Properties)
                 {
-                    IEdmStructuralProperty structuralProperty = this.FindProperty(keyProperty.PropertyName) as IEdmStructuralProperty;
+                    IEdmStructuralProperty structuralProperty = null;
+                    if (!string.IsNullOrEmpty(keyProperty.Alias))
+                    {
+                        string[] keyProperties = keyProperty.PropertyName.Split('/');
+                        for (int i = 0; i < keyProperties.Length; i++)
+                        {
+                            if (i == 0)
+                            {
+                                structuralProperty = this.FindProperty(keyProperties[i]) as IEdmStructuralProperty;
+                                i++;
+                            }
+                            else
+                            {
+                                structuralProperty = this.FindProperty(keyProperties[i]) as IEdmStructuralProperty;
+                                i++;
+                            }
+
+                            if (structuralProperty.Type.Definition.TypeKind == EdmTypeKind.Complex)
+                            {
+                                structuralProperty = structuralProperty.Type.AsComplex().FindProperty(keyProperties[i]) as IEdmStructuralProperty;

```

Comment:
I don't know if this is getting validated somewhere else (maybe in `CsdlPropertyReference`?) but it would be really bad if we went passed the end of `keyProperties` here. Maybe we should check the length is even before we start the loop